commands/daemon/settings.go

```diff
@@ -144,3 +145,35 @@ func (s *SettingsService) Write(ctx context.Context, req *rpc.WriteRequest) (*rp
 	}
 	return &rpc.WriteResponse{}, nil
 }
+
+// Delete removes a key from the config file
+func (s *SettingsService) Delete(ctx context.Context, req *rpc.DeleteRequest) (*rpc.DeleteResponse, error) {
```

Comment:
There is a very similar function implemented in `internal/cli/config/delete.go`: https://github.com/arduino/arduino-cli/blob/master/internal/cli/config/delete.go#L46-L72, it may be replaced with a function call to this one?